openedx/features/course_experience/views/course_home_messages.py

```diff
@@ -124,19 +127,26 @@ def _register_course_home_messages(request, course, user_access, course_start_da
             ),
             title=Text(_('You must be enrolled in the course to see course content.'))
         )
-    if not user_access['is_anonymous'] and not user_access['is_staff'] and not user_access['is_enrolled']:
-        CourseHomeMessages.register_info_message(
-            request,
-            Text(_(
-                u'{open_enroll_link}Enroll now{close_enroll_link} to access the full course.'
-            )).format(
-                open_enroll_link=HTML('<button class="enroll-btn btn-link">'),
-                close_enroll_link=HTML('</button>')
-            ),
-            title=Text(_(u'Welcome to {course_display_name}')).format(
-                course_display_name=course.display_name
+    if not user_access['is_anonymous'] and not user_access['is_staff'] and \
+            not user_access['is_enrolled']:
+        if not course.invitation_only:
+            CourseHomeMessages.register_info_message(
+                request,
+                Text(_(
+                    u'{open_enroll_link}Enroll now{close_enroll_link} to access the full course.'
+                )).format(
+                    open_enroll_link=HTML('<button class="enroll-btn btn-link">'),
+                    close_enroll_link=HTML('</button>')
+                ),
+                title=Text(_(u'Welcome to {course_display_name}')).format(
+                    course_display_name=course.display_name
+                )
+            )
+        else:
+            CourseHomeMessages.register_info_message(
+                request,
+                Text(_(u'You should be enrolled in the course to fully access the contents.')),
```

Comment:
Can this text match the banner text above, so we don't have so many strings to translate?

> You must be enrolled in the course to see course content.

Reply:
Could you clarify please? Do you mean, replace 

`Text(_(u'You should be enrolled in the course to fully access the contents.'))`


with 

`
Text(_(
                    u'{open_enroll_link}Enroll now{close_enroll_link} to access the full course.'
                )).format(
                    open_enroll_link=HTML('<button class="enroll-btn btn-link">'),
                    close_enroll_link=HTML('</button>')
                )
`

Reply:
@DevanR I think @pomegranited meant the text in line no. 128 above i.e `title=Text(_('You must be enrolled in the course to see course content.'))`

Reply:
Ah, ok got it. I've pushed the change.

Reply:
Thanks to @pkulkark for clarifying my ambiguous comment, and to @DevanR  for making the change!